doc/design/diag_attribute_error.md

```diff
@@ -0,0 +1,96 @@
+# Diagnostic Attribute Error
+
+## Motivation
+
+SQLFlow extended SQL syntax to describe an end-to-end machine learning job, for a typical SQL program:
+
+``` python
+SELECT * FROM iris.train
+TO TRAIN DNNClassifier
+WITH
+  model.n_classes = 2,
+  model.hidden_units = [128, 64]
+LABEL class
+INTO my_model;
+
+SELECT * FROM iris.test
+TO PREDICT iris.pred.class
+USING my_model;
+```
+
+SQLFlow compiles the above SQL program into an execution plan and runs it.  As the TRAIN statement above, SQLFlow uses `TO TRAIN CLAUSE`  to train a specific model called  `DNNClassifier`, using `WITH CLAUSE` to configure the training arguments. 
```

Comment:
SQLFlow compiles each statement in the program into an execution plan and executes them.